app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/EateryCard.kt

```diff
@@ -320,51 +353,96 @@ fun EateryCard(
+                                    interactionSource = interactionSource,
+                                    indication = rememberRipple(radius = 9.dp),
+                                    onClick = {
+                                        onFavoriteClick(!isFavorite)
+                                    }
+                                ),
+                            contentDescription = null
+                        )
+                    }
                 }
-                EateryCardPrimaryHeader(eatery = eatery, isCompact = isCompact)
-                EateryCardSecondaryHeader(eatery = eatery, isCompact = isCompact)
+                EateryCardPrimaryHeader(
+                    eatery = eatery,
+                    isCompact = isCompact,
+                    isGridView = isGridView
+                )
+                EateryCardSecondaryHeader(
+                    eatery = eatery,
+                    isCompact = isCompact,
+                    isGridView = isGridView
+                )
+                //TODO comment until backend is done with entree recommendation
+                EateryCardTertiaryHeader(
+                    eatery = eatery,
+                    isCompact = isCompact,
+                    isGridView = isGridView
+                )
             }
         }
     }
 }
 
 @Composable
-fun EateryCardPrimaryHeader(eatery: Eatery, isCompact: Boolean) {
-    if (isCompact) {
-        Row(
-            modifier = Modifier.padding(top = 2.dp),
-            verticalAlignment = Alignment.CenterVertically
-        ) {
-            Icon(
-                Icons.Default.Schedule,
-                contentDescription = null,
-                tint = GrayFive,
-                modifier = Modifier.padding(end = 4.dp, top = 1.dp)
-            )
+fun GridViewFavoriteWidget(
+    modifier: Modifier = Modifier,
+    isFavorite: Boolean,
+    interactionSource: MutableInteractionSource,
+    onClick: () -> Unit
+) {
+    Surface(
+        modifier = modifier
+            .size(40.dp)
+            .clickable(
+                interactionSource = interactionSource,
+                indication = rememberRipple(radius = 20.dp),
+                onClick = onClick
+            ),
+        shape = CircleShape,
+        color = Color.White
+    ) {
+        Icon(
+            imageVector = if (isFavorite) Icons.Filled.Star else Icons.Outlined.StarOutline,
+            tint = if (isFavorite) Yellow else GrayFive,
+            modifier = Modifier
+                .padding(8.dp),
+            contentDescription = null
+        )
+    }
+}
 
-            Text(
-                text = "${eatery.getWaitTimes() ?: "3-5"} min wait",
-                color = GrayFive,
-                style = EateryBlueTypography.subtitle2
-            )
 
-            DotSeparator()
-            EateryMenuSummary(eatery = eatery)
-        }
+@Composable
+fun EateryCardPrimaryHeader(eatery: Eatery, isCompact: Boolean, isGridView: Boolean) {
```

Comment:
Generally instead of commenting out the bodies of Composables, we should just comment out our usages of them. For this one for example, it might seem like this composable does nothing, but it looks like it could output a Row with top padding of `2.dp` which would create unintentional spacing.